tests/user_restricted/test_user_restricted.py

```diff
@@ -82,13 +84,48 @@ def test_retrieve_patient_with_invalid_auth_header(self, headers):
         helpers.check_response_status_code(response, 401)
         helpers.check_response_headers(response, headers)
 
-    def test_retrieve_patient_with_missing_urid_header(self, headers_with_token):
+    def test_user_role_sharedflow_retrieve_patient_with_missing_urid_header(self, headers_with_token):
         self.headers.pop("NHSD-Session-URID")
         response = helpers.retrieve_patient(
-            retrieve[3]["patient"],
+            retrieve[0]["patient"],
             self.headers
         )
-        helpers.check_retrieve_response_body(response, retrieve[3]["response"])
+        helpers.check_response_headers(response, self.headers)
+        response_body = json.loads(response.text)
+        assert response.status_code == 200
+        # check id matches
+        assert response_body["id"] == retrieve[0]["patient"]
+        assert response_body["resourceType"] == "Patient"
+        # check the shape of response
+        assert response_body["address"] is not None
+        assert isinstance(response_body["address"], list)
+
+        assert response_body["birthDate"] is not None
+        assert isinstance(response_body["birthDate"], str)
+        assert len(response_body["birthDate"]) > 1
+
+        assert response_body["gender"] is not None
+        assert isinstance(response_body["gender"], str)
+
+        assert response_body["name"] is not None
+        assert isinstance(response_body["name"], list)
+        assert len(response_body["name"]) > 0
+
+        assert len(response_body["identifier"]) > 0
+        assert isinstance(response_body["identifier"], list)
+
+        assert len(response_body["extension"]) > 0
+        assert isinstance(response_body["extension"], list)
+
+        assert response_body["meta"] is not None
+
+    def test_user_role_sharedflow_invalid_role(self, headers_with_token):
+        self.headers["NHSD-Session-URID"] = "invalid"
+        response = helpers.retrieve_patient(
+            retrieve[8]["patient"],
```

Comment:
Test is failing "IndexError: list index out of range".
Appears that the scenario `retrieve[8]` does not exist within the pds_scenarios.py file. This scenario needs adding.

Reply:
added in scenario for "retireve[8]"